OrbitGl/FrameTrackOnlineProcessor.cpp

```diff
@@ -0,0 +1,50 @@
+#include "FrameTrackOnlineProcessor.h"
+
+#include <limits>
+#include "CaptureData.h"
+#include "TimeGraph.h"
+#include "UserDefinedCaptureData.h"
+
+FrameTrackOnlineProcessor::FrameTrackOnlineProcessor(const CaptureData& capture_data,
+                                                     TimeGraph* time_graph)
+    : time_graph_(time_graph) {
+  const UserDefinedCaptureData& user_defined_capture_data =
+      capture_data.user_defined_capture_data();
+  for (const auto& function : user_defined_capture_data.frame_track_functions()) {
+    const uint64_t function_address = capture_data.GetAbsoluteAddress(function);
+    current_frame_track_functions_.insert(function_address);
+    previous_timestamp_ns_.insert(
+        std::make_pair(function_address, std::numeric_limits<uint64_t>::max()));
+  }
+}
+
+void FrameTrackOnlineProcessor::ProcessTimer(const orbit_client_protos::TimerInfo& timer_info,
+                                             const orbit_client_protos::FunctionInfo& function) {
+  uint64_t function_address = timer_info.function_address();
+  if (current_frame_track_functions_.empty() ||
+      !current_frame_track_functions_.contains(function_address)) {
+    return;
+  }
+  uint64_t previous_timestamp_ns = previous_timestamp_ns_[function_address];
+  if (previous_timestamp_ns == std::numeric_limits<uint64_t>::max()) {
+    previous_timestamp_ns_[function_address] = timer_info.start();
+    return;
+  }
+
+  if (previous_timestamp_ns < timer_info.start()) {
+    orbit_client_protos::TimerInfo frame_timer;
+
+    // TID is meaningless for this timer (start and end can be on two different threads).
```

Comment:
We should probably think of refactoring timers at some point to introduce a class hierarchy for it. It has a couple of fields that are only used for certain types. Also we do have a couple of switches over the type, which can be simplified by using virtual methods or a visitor patter. 
Anyway, this is clearly out of scope for this PR.

Reply:
Yes, it's a bit awkward to put in a meaningless value here. 

Will resolve for now, but let's chat about our options how to handle this in the future.